kata04/kata04.go

```diff
@@ -74,17 +75,25 @@ func getDataFromFile(filename string, columns []int) [][]string {
 
 	output := make([][]string, 0)
 
-	scanner := bufio.NewScanner(file)
-	for scanner.Scan() {
-		columns, err := getColumns(scanner.Text(), columns, "*")
+	fileReader := bufio.NewReader(file)
+	keepReading := true
+
+	for keepReading {
+		line, err := fileReader.ReadString('\n')
+		if err != nil {
+			if err == io.EOF {
+				keepReading = false
```

Comment:
another way to approach this (not necessarily better,  but one less allocation):
```
	for {
		line, err := fileReader.ReadString('\n')
		if err != nil {
			if err == io.EOF {
				break
```